src/site/content/en/blog/optimize-long-tasks/index.md

```diff
@@ -0,0 +1,403 @@
+{% endAside %}
+
+When tasks are broken up, other tasks can be prioritized better by the browser's internal prioritization scheme. One way to yield to the main thread involves using a combination of a `Promise` that resolves with a call to `setTimeout`:
+
+```js
+function yieldToMain () {
+  return new Promise(resolve => {
+    setTimeout(resolve, 0);
+  });
+}
+```
+
+{% Aside 'caution' %}
+While this code example returns a `Promise` that resolves after a call to `setTimeout`, it's not the `Promise` that is responsible for running the rest of the code in a new task, it's the `setTimeout` call. Promise callbacks run as microtasks rather than tasks, and therefore don't yield to the main thread.
+{% endAside %}
+
+Then in the `saveSettings` function, you can yield to the main thread after each bit of work if you `await` the `yieldToMain()` function after each function call:
+
+```js
+async function saveSettings () {
+  // Create an array of functions to run:
+  const tasks = [
+    validateForm,
+    showSpinner,
+    saveToDatabase,
+    updateUI,
+    sendAnalytics
+  ]
+
+  // Loop over the tasks:
+  while (tasks.length > 0) {
+    // Shift the first task off the tasks array:
+    const task = tasks.shift();
+
+    // Run the task:
+    task();
+
+    // Yield to the main thread:
+    await yieldToMain();
+  }
+}
+```
+
+{% Aside 'important' %}
+You don't have to yield after _every_ function call. For example, if you run two functions that result in critical updates to the user interface, you probably don't want to yield in between them. If you can, let that work run _first_, then consider yielding in between functions that do less critical or background work that the user doesn't see.
+{% endAside %}
+
+The result is that the once-monolithic task is now broken up into separate tasks.
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/wg0FW6S29CzOCbbwk9kK.png", alt="The same saveSettings function depicted in Chrome's performance profiler, only with yielding. The result is the once-monolithic task is now broken up into five separate tasks&mdash;one for each function.", width="800", height="211" %}
+  <figcaption>
+    The <code>saveSettings</code> function now executes its child functions as separate tasks.
+  </figcaption>
+</figure>
+
+The benefit of using a promise-based approach to yielding rather than manual use of `setTimeout` is better ergonomics. Yield points become declarative, and therefore easier to write, read, and understand.
+
+### Yield only when necessary
+
+What if you have a bunch of tasks, but you only want to yield if the user attempts to interact with the page? That's the kind of thing that [`isInputPending`](/isinputpending/) was made for!
+
+`isInputPending` is a function you can run at any time to determine if the user is attempting to interact with a page element: a call to `isInputPending` will return `true`. It returns `false` otherwise.
```

Comment:
```suggestion
`isInputPending()` is a function you can run at any time to determine if the user is attempting to interact with a page element: a call to `isInputPending()` will return `true`. It returns `false` otherwise.
```